src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2DPaintApplier.java

```diff
@@ -828,6 +841,37 @@ private COSArray buildType2Functions(List<Color> colors, COSArray domain, COSArr
         }
     }
 
+    /**
+     * Get the value of a private field from the specified object.
+     *
+     * @param obj The object containing the field.
+     * @param fieldName The name of the field.
+     * @param <T> The field type.
+     * @return The private field value.
+     */
+    @SuppressWarnings({ "unchecked", "WeakerAccess" })
+    protected static <T> T getPrivateFieldValue(Object obj, String fieldName)
+    {
+        try
+        {
+            Class<?> c = obj.getClass();
+            try
+            {
+                Field f = c.getDeclaredField(fieldName);
+                f.setAccessible(true);
+                return (T)f.get(obj);
+            }
+            catch (NoSuchFieldException ignored)
+            {
+            }
+            throw new NullPointerException("Field " + fieldName + " not found!");
```

Comment:
Not rethrowing the underlying NoSuchFieldException seems strange. The method getPropertyValue() you likely copied here did loop, and ignored that error because of that. When accessing a private field you in theory also need to loop.

Reply:
You're right, it should throw and I've changed it back to use a loop as you suggest. The theory was that a private field is only accessible from the class that contains it, but we've already broken encapsulation.